### ISSUE TITLE
Convert all HTML tags in translation strings to use createInterpolateElement

### ISSUE THREAD

**Issue by tofumatt** [google/site-kit-wp#2216]
## Feature Description

We currently use `<span class="screen-reader-text">` and `<a>` tags throughout our app inside i18n strings (example: https://github.com/google/site-kit-wp/blob/af48c90e7eb72a700fd15e951491731f08758c4c/assets/js/components/setup/compatibility-checks.js#L139).

This is limiting because it means we can only use raw HTML and can't use React components in places we might like to. Instead of this existing approach, we should interpolate tags using actual React components and `createInterpolateElement`. We do this already in some places (see the ACs for more info).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* All HTML tags (eg `<span>` or `<a>` tags) in any of our i18n strings should be converted into actual components—they should be using `createInterpolateElement` like we do here: https://github.com/google/site-kit-wp/blob/af48c90e7eb72a700fd15e951491731f08758c4c/assets/js/components/source-link.js#L45-L59
* There should be no `<span>` tags with any attributes in i18n strings.
* Existing i18n strings with `<span>` tags that relied on `dangerouslySetInnerHTML` (eg https://github.com/google/site-kit-wp/blob/af48c90e7eb72a700fd15e951491731f08758c4c/assets/js/modules/pagespeed-insights/components/common/MetricsLearnMoreLink.js#L36-L42) should no longer use `dangerouslySetInnerHTML`

## Implementation Brief

In the following files:
- /assets/js/components/error-text.js
- /assets/js/components/optin.js
- /assets/js/components/reset-button.js
- /assets/js/components/notifications/notification.js
- /assets/js/components/dashboard-details/DashboardDetailsEntityNotFoundView.js
- /assets/js/modules/optimize/components/common/UseSnippetInstructions.js
- /assets/js/modules/pagespeed-insights/components/settings/SettingsView.js
- /assets/js/modules/pagespeed-insights/components/common/ReportDetailsLink.js

Any props set using `dangerouslySetInnerHTML` and `sanitizeHTML` functions should be refactored to use `createInterpolateElement`

- The following files will be updated to not use dangerouslySetInnerHTML when the [following PR](https://github.com/google/site-kit-wp/pull/2163) will be merged and should not be updated.
	- /assets/js/components/Link.js
	- /assets/js/components/setup/compatibility-checks.js
	- /assets/js/modules/adsense/components/setup/SetupAccountCreate.js
	- /assets/js/modules/adsense/components/setup/SetupAccountNoClient.js
	- /assets/js/modules/pagespeed-insights/components/common/MetricsLearnMoreLink.js
	- /assets/js/modules/analytics/components/common/AnonymizeIPSwitch.js

### Test Coverage

* 

### Visual Regression Changes

* 

## QA Brief

* 

## Changelog entry

*

**Comment by felixarntz** [comment]
As far as I can tell, this issue is indeed no longer relevant after #2163. Closing.
